sys/include/benchmark.h

```diff
@@ -44,6 +44,10 @@ extern "C" {
 #define BENCHMARK_FUNC(name, runs, func)                        \
     do {                                                        \
         ztimer_stopwatch_t timer = { .clock = ZTIMER_USEC };    \
+        /* warm up cache, branch predictor, ... */              \
```

Comment:
Maybe keep this optional? Maybe:

```
#define BENCHMARK_FUNC_WARMUP(name, runs, warmup, func)
//...
#define BENCHMARK_FUNC(name, runs, func) BENCHMARK_FUNC_WARMUP(name, runs, 0, func)
```


Mostly because *sometimes* 32 runs might already be a lot (e.g., when testing `O(n^2)` algos on unrealistically large sets). And we don't want to make this adaptive here (like, specify warmup in milliseconds) I guess.